backend/src/database/user_catalogue.go

```diff
@@ -46,33 +51,53 @@ type UserPrograms struct {
 	TotalTime      uint    `json:"total_time"`
 }
 
-func (db *DB) GetUserPrograms(userId uint, order string, orderby string, search string, tags []string) ([]UserPrograms, uint, uint, error) {
+func validOrder(str string) string {
+	if slices.Contains([]string{"asc", "desc"}, strings.ToLower(str)) {
+		return strings.ToLower(str)
+	}
+	return "desc"
+}
+
+func (db *DB) GetUserPrograms(userId uint, order string, orderBy string, search string, tags []string) ([]UserPrograms, uint, uint, error) {
 	programs := []UserPrograms{}
-	orderStr := orderby + " " + order
+	fieldMap := map[string]string{
+		"program_name":    "p.name",
+		"provider_name":   "pp.name",
+		"course_progress": "course_progress",
+		"is_favorited":    "is_favorited",
+	}
+	dbField, ok := fieldMap[orderBy]
+	if !ok {
+		dbField = "p.name"
+	}
+	orderStr := dbField + " " + validOrder(order)
 	tx := db.Conn.Table("programs p").
 		Select(`p.id, p.thumbnail_url,
     p.name as program_name, pp.name as provider_name, p.external_url,
     f.user_id IS NOT NULL as is_favorited,
-    CASE WHEN COUNT(o.type) > 0 THEN 100 ELSE COUNT(milestones.id) * 100.0 / p.total_progress_milestones END as course_progress,
-	a.total_time`).
+    CASE
+        WHEN EXISTS (SELECT 1 FROM outcomes o WHERE o.program_id = p.id AND o.user_id = ?) THEN 100
```

Comment:
fixes the div by 0 error, but also returns the appropriate amount of programs as mentioned